check_license.sh

```diff
@@ -48,8 +48,37 @@ then
     cd "$1"
 fi
 
-# Known licenses must continue to match.
-shasum -a 256 -c $CHKSUM_FILE
+# Remove all newlines from the Checksum file as these are reported as formatting
+# errors by the shasum program
+TMP_CHKSUM_FILE=$(mktemp)
+trap cleanup EXIT
+cleanup()
+{
+  rm $TMP_CHKSUM_FILE
+}
+sed '/^$/d' $CHKSUM_FILE > $TMP_CHKSUM_FILE
+
+# Use the tmp-file for the rest of the script
+CHKSUM_FILE=$TMP_CHKSUM_FILE
+
+# Collect only stderr from the subcommand
+output="$(exec 3>&1; shasum --warn --algorithm 256 --check $CHKSUM_FILE >/dev/null 2>&3)"
+
+if ! grep -q 'lines are improperly formatted' 2>/dev/null < "$output"; then
+  echo >&2 "Some lines in the LIC_FILE_CHKSUM.sha256 file are misformed:"
+  (
+    IFS="
+"
```

Comment:
Can you help me understanding why we need this?

Reply:
It's just for word splitting in the following for loop to only split on newlines

Reply:
Ah, of course. Thanks